cmd/argo/commands/cp.go

```diff
@@ -0,0 +1,145 @@
+package commands
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	"github.com/argoproj/argo-workflows/v3/cmd/argo/commands/client"
+	"github.com/argoproj/argo-workflows/v3/pkg/apiclient"
+	workflowpkg "github.com/argoproj/argo-workflows/v3/pkg/apiclient/workflow"
+	"github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+)
+
+type copyArtifactOpts struct {
+	namespace    string // --namespace
+	nodeId       string // --node-id
+	templateName string // --template-name
+	artifactName string // --artifact-name
+}
+
+func NewCpCommand() *cobra.Command {
+	var copyArtifactArgs copyArtifactOpts
+
+	command := &cobra.Command{
+		Use:   "cp my-wf output-directory ...",
+		Short: "copy artifacts from workflow",
+		Example: `# Copy a workflow's artifacts to a local output directory:
+
+  argo cp my-wf output-directory
+
+# Copy artifacts from a specific node in a workflow to a local output directory:
+
+  argo cp my-wf output-directory --node-id=my-wf-node-id-123
+`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 2 {
+				cmd.HelpFunc()(cmd, args)
+				return fmt.Errorf("incorrect number of arguments")
+			}
+			workflowName := args[0]
+			outputDir := args[1]
+
+			argoServerUrl, err := cmd.Parent().Flags().GetString("argo-server")
```

Comment:
change `argoServerOpts = apiclient.ArgoServerOpts{}` to `ArgoServerOpts = apiclient.ArgoServerOpts{}` then you can access that